torch/_functorch/aot_autograd.py

```diff
@@ -3167,9 +3189,12 @@ def create_aot_dispatcher_function(
         enable_python_dispatcher() if shape_env is not None else nullcontext()
     )
 
-    with torch.autograd.set_multithreading_enabled(
-        False
-    ), preserve_rng_state(), fake_mode, python_dispatcher_mode, PhiloxStateTracker():
+    with ExitStack() as stack:
+        stack.enter_context(torch.autograd.set_multithreading_enabled(False))
+        stack.enter_context(preserve_rng_state())
+        stack.enter_context(fake_mode)
+        stack.enter_context(python_dispatcher_mode)
+        stack.enter_context(PhiloxStateTracker())
```

Comment:
hmm. It would be nice if we can use the normal `with ...` syntax for all of our context managers. Why did we need `ExitStack`?

Reply:
So I think the reason I changed to use `ExitStack ` is because I want to be able to enter context for assertion functionalization programmatically instead of always on. How about I change it to something like:
```
with torch.autograd.set_multithreading_enable(...), ..., ExitStack() as stack:
    ...
    if FunctionalAssertionsHelper.can_functionalize(...):
        stack.enter(FunctionalAssertionsHelper.get_state_tracker())
```

Reply:
ah ok - the standard pattern for doing that that I've seen throughout the codebase is:
```
maybe_enable_functionalize_asserts = nullcontext()
if FunctionalAssertionsHelper.can_functionalize(...):
    maybe_enable_functionalize_asserts = FunctionalAssertionsHelper.get_state_tracker()
...

with maybe_enable_functionalize_asserts:
```

Reply:
Sure can do that. I guess the downside of it is that it adds another level of indent though :p